### PR TITLE
[DOC] Add datasets description to the documentation

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Relates to #4218 

Changes proposed in this pull request:

- render RST files containing each "dataset" as part of the official doc
 - the files are symlinked into the doc folder before any doc build
- update bibliography of each example
- general clean of the dataset descriptions (links, headers, link to glossary...)

TODO

- [x] update changelog
- [x] link each description to its fetcher function or an example of its usage
- [x] add reference for each page (to make it easier to link to)
- [x] mention bids_langloc is a duplicate

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Dataset description in the doc:

https://output.circle-artifacts.com/output/job/c1db91cc-2ac2-43e8-903b-c98b84dcb77c/artifacts/0/dev/datasets/index.html

@bthirion @ymzayek @emdupre @jeromedockes @htwangtw 

Need some quick opinions before I keep working on this.

- do we think it is a good idea to have the descriptions of the atlas, templates, datasets as part of the doc?
- if so where should it go? as a standalone section like right now? along with the dataset module as part of the API section (https://nilearn.github.io/dev/modules/datasets.html)? Something else?

**Comment by ymzayek** [comment]
@Remi-Gau I like having this information all there in one spot and easily accessible however since dataset fetching is not a core functionality and really supposed to be only to allow us to showcase our features I think putting it there in the left hand side navigation bar gives it too much importance. Maybe including just in https://nilearn.github.io/dev/modules/datasets.html would be ok

File: nilearn/datasets/description/aal_SPM12.rst [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+=========
 
 
 Notes
 -----
-This atlas is the result of an automated anatomical parcellation of the spatially normalized single-subject high-resolution T1 volume provided by the Montreal Neurological Institute (MNI) (D. L. Collins et al., 1998, Trans. Med. Imag. 17, 463-468, PubMed).
+This atlas is the result of an automated anatomical :term:`parcellation`
+of the spatially normalized single-subject high-resolution T1 volume
+provided by the Montreal Neurological Institute (MNI)
+(D. L. Collins et al., 1998, Trans. Med. Imag. 17, 463-468, PubMed).
```
Comment (bthirion) [inline nilearn/datasets/description/aal_SPM12.rst:10]: make it formally a citation  ?

**Review by bthirion — COMMENTED** [review]
LGTM otherwise.

File: nilearn/datasets/description/aal_SPM12.rst (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/datasets/description/aal_SPM12.rst:10]: Actually there is a whole bunch in of other references mentioned that are not in any of our .bib files so I was planning to tackle those in a separate PR.

But let me know if you me to do it in this one.

Comment (bthirion) [reply nilearn/datasets/description/aal_SPM12.rst:10]: This PR is already pretty big. It's OK to me if you address this in another PR.
Best,

File: nilearn/datasets/description/bids_langloc.rst [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
+.. _bids_langloc_dataset:
 
+BIDS language localizer
+=======================
 
-Content
--------
+.. warning::
 
+    This dataset is a duplicate of :ref:`langugage_localizer_dataset`.
```
Comment (bthirion) [inline nilearn/datasets/description/bids_langloc.rst:8]: Hm, there's a typo here, but the link is correct ?

File: nilearn/datasets/description/language_localizer_demo.rst [hunk 3]
```diff
@@ -1,5 +1,11 @@
-language localizer demo
+.. _langugage_localizer_dataset:
```
Comment (bthirion) [inline nilearn/datasets/description/language_localizer_demo.rst:1]: Here is the typo ;-)

**Review by bthirion — COMMENTED** [review]
LGTM otherwise, thx.

File: nilearn/datasets/description/language_localizer_demo.rst (hunk 3 above)
Comment (Remi-Gau) [reply nilearn/datasets/description/language_localizer_demo.rst:1]: ha thanks!!

**Review by bthirion — APPROVED** [review]

**Pull request merged**